Automattic/wp-calypso
# Require non-empty password to publish password protected posts

Fixes #16344 

This PR makes the password for password-protected publishing required, thus prevents the users from inadvertently publishing public posts by using an empty password.

![password-protected-15fps](https://user-images.githubusercontent.com/1017839/28489740-98762310-6eca-11e7-87e7-af886629ac63.gif)

This is part of a larger epic (See p3Ex-2ri-p2).

To test:
- Checkout this branch
- Run the unit tests: `npm run test-client client/state/posts/test`
- Run this branch with the publish-confirmation feature enabled: `ENABLE_FEATURES=post-editor/delta-post-publish-flow make run`
  - Draft a post
  - Change the visibility to Password Protected under Post Settings --> Status
  - Verify that the password field is empty, it has the focus, no error message is shown yet, and the Publish... button is disabled
  - Click outside the password input field and verify that an error message is shown
  - Type a password and verify that the error message is gone and the Publish... button is enabled
  - Delete the password and verify that after deleting the last character the error message is displayed again and the Publish... button gets disabled
  - Change the visibility back to Public and click Publish...
  - On the confirmation sidebar verify that the Publish! button is enabled, then change the visibility of the post from Public to Password Protected
  - Verify that the behavior is the same as on the editor sidebar in the previous steps as you type/delete password and take away the focus from the field when it's empty
- Run this branch without the publish-confirmation feature enabled: `make run` and verify that the original flow has not been affected (same as master)
